harper-core/src/parsers/collapse_identifiers.rs

```diff
@@ -1,24 +1,24 @@
-use crate::Lrc;
 use std::collections::VecDeque;
+use std::sync::Arc;
 
 use itertools::Itertools;
 
 use super::{Parser, TokenKind};
 use crate::patterns::{PatternExt, SequencePattern};
-use crate::{Dictionary, FullDictionary, MergedDictionary, Span, Token, VecExt};
+use crate::{Dictionary, Lrc, Span, Token, VecExt};
 
 /// A parser that wraps any other parser to collapse token strings that match
 /// the pattern `word_word` or `word-word`.
 pub struct CollapseIdentifiers {
     inner: Box<dyn Parser>,
-    dict: Lrc<MergedDictionary<FullDictionary>>,
+    dict: Arc<dyn Dictionary>,
 }
 
 impl CollapseIdentifiers {
-    pub fn new(inner: Box<dyn Parser>, dict: &Lrc<MergedDictionary<FullDictionary>>) -> Self {
+    pub fn new(inner: Box<dyn Parser>, dict: Box<Arc<dyn Dictionary>>) -> Self {
```

Comment:
Is there a reason this is an `Arc` instead of an `Lrc`?

Reply:
I don't remember exactly why, but there were issues with `MergedDictionary` and using `Lrc`. [Here's the commit in question if you want to find another way](https://github.com/elijah-potter/harper/pull/258/commits/ee77e5d9ab66bfdbcf35fdeba8a90892be8f9145)

Reply:
If I remember correctly, it had to do with the `flat_map` in `MergedDictionary`'s `words_iter()` function.